### PR TITLE
Resecurity pack

### PR DESCRIPTION
**Pull request by oleyur**
## Partner ID:
resecurity-15606

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
New pack contributing.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by GuyAfik** [comment]
> [CLA assistant check](https://cla-assistant.io/demisto/content?pullRequest=26071) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=26071) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=26071) it.

@oleyur can you please sign the CLA?

**Comment by oleyur** [comment]
> Hi @oleyur - any updates on this? Looks like some jobs are failing (Run `./Utils/git_pull_master_into_fork.sh` or merge manually from upstream `demisto/content` to fix the pre-commit job failure), and we're awaiting for you to sign the contribution form.

Hello, I have merged updates from upstream, check please.

**Comment by oleyur** [comment]
> > > Hi @oleyur - any updates on this? Looks like some jobs are failing (Run `./Utils/git_pull_master_into_fork.sh` or merge manually from upstream `demisto/content` to fix the pre-commit job failure), and we're awaiting for you to sign the contribution form.
> > 
> > 
> > Hello, I have merged updates from upstream, check please.
> 
> Thanks, please make sure to sign the CLA before we can start the review process: [#26071 (comment)](https://github.com/demisto/content/pull/26071#issuecomment-1520124936)

We have signed the CLA. Please check.

**Comment by kgal-pan** [comment]
@oleyur - Looks like the [code coverage is lower than threshold (70%)](https://app.circleci.com/pipelines/github/demisto/content/256483/workflows/ff6c8cd0-914f-4cb2-be92-ea030dff749c/jobs/698645?invite=true#step-121-27). Please increase it by reviewing the missed lines using [this report](https://output.circle-artifacts.com/output/job/6b008b65-b28e-408d-bfcb-310811ff608b/artifacts/0/artifacts/coverage_report/html/d_0a0fa62b3d48dea6_ResecurityMonitoring_py.html).

Also, [this job](https://app.circleci.com/pipelines/github/demisto/content/256483/workflows/ff6c8cd0-914f-4cb2-be92-ea030dff749c/jobs/698647) failing for found secrets. Please address.

**Comment by oleyur** [comment]
> @oleyur - Looks like the [code coverage is lower than threshold (70%)](https://app.circleci.com/pipelines/github/demisto/content/256483/workflows/ff6c8cd0-914f-4cb2-be92-ea030dff749c/jobs/698645?invite=true#step-121-27). Please increase it by reviewing the missed lines using [this report](https://output.circle-artifacts.com/output/job/6b008b65-b28e-408d-bfcb-310811ff608b/artifacts/0/artifacts/coverage_report/html/d_0a0fa62b3d48dea6_ResecurityMonitoring_py.html).
> 
> Also, [this job](https://app.circleci.com/pipelines/github/demisto/content/256483/workflows/ff6c8cd0-914f-4cb2-be92-ea030dff749c/jobs/698647) failing for found secrets. Please address.

Hello, secrets related changes and code coverage changes pushed. Please check.

**Review by GuyAfik — CHANGES_REQUESTED** [review]
@oleyur incredible work, please see my comments:

1. great unit-tests, well done!
2. you can remove the `Pipfile` and `Pipfile.lock` that gets created automatically, those files are irrelevant.
3. let me know if you need help with anything, feel free to slack me up in DFIR.

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    """_summary_
+
+    Args:
+        module_name (str): _description_
+    """
+    def get_task_monitor_results(client: Client, args: Dict) -> CommandResults:
+
+        # get params from user
+        monitor_task_id = arg_to_number(args.get("monitor_task_id"), 'monitor_task_id', True)
+        limit = int(args.get("limit", DEFAULT_RESULTS_SIZE_LIMIT))
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    # if your Client class inherits from BaseClient, SSL verification is
+    # handled out of the box by it, just pass ``verify_certificate`` to
+    # the Client constructor
+    verify_certificate = not demisto.params().get('insecure', False)
+
+    # if your Client class inherits from BaseClient, system proxy is handled
+    # out of the box by it, just pass ``proxy`` to the Client constructor
+    proxy = demisto.params().get('proxy', False)
+
+    demisto.debug(f'Command being called is {demisto.command()}')
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:223]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            # This is the call made when pressing the integration Test button.
+            result = test_module(client)
+            return_results(result)
+
+        else:
+            return_results(commands[command](client, args))
+
+    # Log exceptions and return errors
+    except Exception as e:
+        return_error(f'Failed to execute {demisto.command()} command.\nError:\n{str(e)}')
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:251]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    outputs:
+    - contextPath: Resecurity.DataBreach.id
+      description: 'Data breach ID'
+      type: String
+    - contextPath: Resecurity.DataBreach.query
+      description: 'The query for which the result was found'
+      type: String
+    - contextPath: Resecurity.DataBreach.detection_date
+      description: 'Data breach detection date'
+      type: int
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.yml]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+STATUS_LIST_TO_RETRY = [429, 500]
+
+STATUS_CODE_MSGS = {
+    401: "Bad Credentials",
+    403: "Forbidden. Something is wrong with your account, please, contact to Resecurity.",
+    404: "Not found. There is no such data on server.",
+    500: "There are some troubles on server with your request."
+}
+
+DEFAULT_RESULTS_SIZE_LIMIT = 1000
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+        return response
+
+
+''' HELPER FUNCTIONS '''
+
+
+def get_human_readable_output(module_name, monitor_task_id, result):
+    return tableToMarkdown(name="{0} results from task with ID {1}".format(module_name, monitor_task_id),
+                           t=result, removeNull=False, date_fields=['detection_date'])
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+            "get-task-monitor-results-data-breaches": get_task_monitor_results_command(MODULE_NAME_BREACHES)
+        }
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration Test button.
+            result = test_module(client)
+            return_results(result)
+
+        else:
+            return_results(commands[command](client, args))
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:247]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+    message: str = ''
+    try:
+        result = client.check_connection()
+        message = result['message']
+    except DemistoException as e:
+        if 'Forbidden' in str(e) or 'Authorization' in str(e):
+            message = 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+    return message
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:131]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+            auth=(api_key, ''),
+            headers=headers,
+            proxy=proxy)
+
+        # command cases implementation
+        commands = {
+            "get-task-monitor-results-data-breaches": get_task_monitor_results_command(MODULE_NAME_BREACHES)
+        }
+
+        if demisto.command() == 'test-module':
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+                page += 1
+
+                if page > total_pages:
+                    break
+
+        # compose result data
+        return CommandResults(
+            outputs_prefix="Resecurity.{0}".format(MAPPING.get(module_name, {}).get("prefix", "").replace(" ", "")),
+            outputs_key_field="id",
+            outputs=result,
```
[the remark this answers is not in the document]
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py]: done

**Comment by oleyur** [comment]
> @oleyur incredible work, please see my comments:
> 
> 1. great unit-tests, well done!
> 2. you can remove the `Pipfile` and `Pipfile.lock` that gets created automatically, those files are irrelevant.
> 3. let me know if you need help with anything, feel free to slack me up in DFIR.

@GuyAfik Hello, thank you for your suggestions. Done

**Comment by GuyAfik** [comment]
@oleyur can you add few more unit-tests to achieve the code coverage of at least 70%? https://app.circleci.com/pipelines/github/demisto/content/257008/workflows/f31c8178-2720-4bcc-9342-c2213598d25c/jobs/699040

other than that looks good, lets schedule a demo.

**Comment by oleyur** [comment]
> @oleyur can you add few more unit-tests to achieve the code coverage of at least 70%? https://app.circleci.com/pipelines/github/demisto/content/257008/workflows/f31c8178-2720-4bcc-9342-c2213598d25c/jobs/699040
> 
> other than that looks good, lets schedule a demo.

of course, just updated.

**Comment by GuyAfik** [comment]
> > @oleyur can you add few more unit-tests to achieve the code coverage of at least 70%? https://app.circleci.com/pipelines/github/demisto/content/257008/workflows/f31c8178-2720-4bcc-9342-c2213598d25c/jobs/699040
> > other than that looks good, lets schedule a demo.
> 
> of course, just updated.

excellent, thank you, sent you an email to schedule a demo, let me know there your availability

**Comment by kgal-pan** [comment]
Hi @oleyur - We're going to close this PR as there have been numerous attempts to contact the ReSecurity team via email and have received no response in 3 weeks. Once a response has been received, we can reopen this PR (or open a new one if needed) and set up a demo.

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.yml [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+    - default: false
+      description: "Affects which results will be included in dataset. Possible values: 1 - only new results, 2 - last results (default value), 3 - all results"
+      isArray: false
+      name: mode
+      required: false
+      secret: false
+    deprecated: false
+    description: 'Retrieve monitoring results from a specific monitor task'
+    execution: false
+    name: get-task-monitor-results-data-breaches
```
Comment (kgal-pan) [inline Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.yml:64]: Please add `resecurity-` prefix to the command name.

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 12]
```diff
@@ -0,0 +1,260 @@
...
+                result += response.json()
+                result_count = len(result)
+                page += 1
+
+                if page > total_pages:
+                    break
+
+        # compose result data
+        return CommandResults(
+            outputs_prefix="Resecurity.{0}".format(MAPPING.get(module_name, {}).get("prefix", "").replace(" ", "")),
```
Comment (GuyAfik) [inline Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:184]: please use f strings

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.yml (hunk 11 above)
Comment (GuyAfik) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.yml:64]: +1 @oleyur please handle it and update code accordingly

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+
+            result = []
+            while limit is not None and result_count < limit:
+                response = client.get_task_monitor_results(monitor_task_id, module_name, page, page_size, mode)
+
+                total_pages = response.headers.get(PAGINATION_HEADER_NAME)
+                if not total_pages:
+                    demisto.debug(total_pages)
+                    raise DemistoException(
+                        "Something is wrong, header {0} is empty for API request".format(PAGINATION_HEADER_NAME)
```
Comment (GuyAfik) [inline Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:171]: use f strings

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py [hunk 14]
```diff
@@ -0,0 +1,260 @@
...
+            timeout=TIMEOUT, retries=RETRIES, status_list_to_retry=STATUS_LIST_TO_RETRY
+        )
+
+        # check response status
+        if response.status_code != 200:
+            if response.status_code in STATUS_CODE_MSGS:
+                raise DemistoException(STATUS_CODE_MSGS[response.status_code])
+            else:
+                raise DemistoException(
+                    "Status code {0} for API request".format(response.status_code)
```
Comment (GuyAfik) [inline Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:88]: please use f strings

**Review by GuyAfik — COMMENTED** [review]
@oleyur looks very good post demo, please fix the last comments, merge from master to trigger the build again.

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py (hunk 13 above)
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:171]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py (hunk 14 above)
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:88]: done

File: Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py (hunk 12 above)
Comment (oleyur) [reply Packs/Resecurity/Integrations/ResecurityMonitoring/ResecurityMonitoring.py:184]: done

**Review by GuyAfik — APPROVED** [review]
@oleyur looks good, thank you for your efforts on this one!

**Pull request merged**